### DISCUSSION TITLE
Testing mui5 branch

### DISCUSSION THREAD

**Discussion by lduarte1991** [ProjectMirador/mirador#3792] [General]
Been taking a look at the mui5 branch and was pointed to the discussion section to clarify some things before making PRs.

1. I had been working on updating some tests, which I'll be adding to my [test branch](https://github.com/lduarte1991/mirador/tree/mui5-tests-luis) and in [one of the tests ](https://github.com/lduarte1991/mirador/commit/b54eff343095c2dc6bd68d8acaf2488a1a9ab1c9) I noticed one of the css names had changed and the `mirador-` classname prefix had been removed. I do not know if the appropriate solution here would be to a) update the test and make a PR or b) open an issue in case the solution *should* have the prefix and the test should not be changed.
2. About 20 failing tests in `OpenSeadragonViewer.test.js` were due to the line `OpenSeadragon.getViewer`. After some quick research, it seems like the function `getViewer` did not exist back in `openseadragon@2.4.2` even though the package.json says it's acceptable (and also this is what a clean `npm install` decided to install on my end). Once I did a manual upgrade to `openseadragon@4.1.0`, all the tests passed for me. Similar to the first point above, should this be a discussion about a) removing support for `openseadragon@2.4.2` or b) updating the tests (if possible) to have a way to be backwards compatible with older openseadragon since we are saying that 2.4.2 is acceptable to use. 

Hope this is all helpful to getting to React 18! will update with more finds/questions as they arise.

**Comment by lduarte1991** [comment]
A large number of the tests are also failing since there's an expectation of certain classes (horizontal, vertical, and small being the most obvious) that do not get added to elements anymore. Obviously, while removing them would fix the tests, I'm assuming it would be more helpful for someone who knows what the styling should expect to happen to add more meaningful tests. 

As an example of what I mean, horizontal had the expectation that the width would be set to 100% so a test like the following might work:
```
const style = getComputedStyle(container.querySelector('.companion-area-bottom'));
expect(style.width).toBe('100%')
```

**Comment by lutzhelm** [comment]
PR #3798 addresses the failure of `OpenSeadragonViewer.test.js` with 2.4.2. The unavailable function is not used outside of this test case.

**Comment by whunter** [comment]
Hi, 
A while back I was asking how people are implementing Mirador right now, given that 3.3.0 causes "one or two" dependency conflicts with newer versions of (other stuff). @enriquediaz suggested that I build the mui5 branch and try it out and I finally got a minute to do so. So here's what I'm seeing. I was able to update our overall stack considerably with the mui5 branch and it builds fine. It's loading the manifest and rendering materials but the config seems like it's changed maybe? Here's some actual examples with the same manifest/tiles.
Running `mui5` branch:
https://update-errthing.d1heayih4ib7xo.amplifyapp.com/archive/fp794v03
Same record running on 3.3.0:
https://iawa.lib.vt.edu/archive/9b02j94t

Any idea what I might need to refactor? Or is this just because the branch is still in development?

Thanks!

**Comment by marlo-longley** [comment]
@whunter Hello! Unfortunately we are just getting to audit things under Discussions, this area was not seen for a while. I assume this is stale as MUI 7 (with 9 on the way) is out in Mirador 4. Thank you for your testing. Did you resolve these issues? Please post under Issues if we can help you with anything else.
